src/java_bytecode/java_bytecode_parser.cpp

```diff
@@ -1381,6 +1390,124 @@ void java_bytecode_parsert::rclass_attribute(classt &parsed_class)
+    {
+      u2 bootstrap_methodhandle_ref = read_u2();
+      const pool_entryt &entry = pool_entry(bootstrap_methodhandle_ref);
+      u2 num_bootstrap_arguments = read_u2();
+
+      lambda_method_handlet handle;
+      status() << "INFO: parse BootstrapMethod handle "
+               << num_bootstrap_arguments << " #args"
+               << eom;
+      parse_methodhandle(entry, handle);
+      if(
+        handle.handle_type ==
+          method_handle_typet::BOOTSTRAP_METHOD_HANDLE_ALT ||
+        handle.handle_type == method_handle_typet::BOOTSTRAP_METHOD_HANDLE)
+      {
+        // try parsing bootstrap method handle
+        if(num_bootstrap_arguments >= 3)
+        {
+          // each entry contains a MethodHandle structure
+          // u2 tag
+          // u2 reference kind which must be in the range from 1 to 9
+          // u2 reference index into the constant pool
+          //
+          // reference kinds use the following
+          // 1 to 4 must point to a CONSTANT_Fieldref structure
+          // 5 or 8 must point to a CONSTANT_Methodref structure
+          // 6 or 7 must point to a CONSTANT_Methodref or
+          // CONSTANT_InterfaceMethodref structure, if the class file version
+          // number is 52.0 or above, to a CONSTANT_Methodref only in the case
+          // of less than 52.0
+          // 9 must point to a CONSTANT_InterfaceMethodref structure
+
+          // the index must point to a CONSTANT_String
+          //                           CONSTANT_Class
+          //                           CONSTANT_Integer
+          //                           CONSTANT_Long
+          //                           CONSTANT_Float
+          //                           CONSTANT_Double
+          //                           CONSTANT_MethodHandle
+          //                           CONSTANT_MethodType
+          u2 arg_index1 = read_u2();
+          u2 arg_index2 = read_u2();
+          u2 arg_index3 = read_u2();
+
+          // skip rest
+          for(size_t i = 3; i < num_bootstrap_arguments; i++)
+            read_u2();
+
+          const pool_entryt &arg1 = pool_entry(arg_index1);
+          const pool_entryt &arg2 = pool_entry(arg_index2);
+          const pool_entryt &arg3 = pool_entry(arg_index3);
+
+          if(!(arg1.tag == CONSTANT_MethodType &&
+               arg2.tag == CONSTANT_MethodHandle &&
+               arg3.tag == CONSTANT_MethodType))
+            return;
+
+          lambda_method_handlet real_handle;
+          status() << "INFO: parse lambda handle" << eom;
+          const pool_entryt &lambda_entry = pool_entry(arg_index2);
+          parse_methodhandle(lambda_entry, real_handle);
+          if(
+            real_handle.handle_type !=
+            method_handle_typet::LAMBDA_METHOD_HANDLE)
+          {
+            lambda_method_handlet empty_handle;
+            parsed_class.lambda_method_handle_map[parsed_class.name].push_back(
+              empty_handle);
```

Comment:
If there are cases where we may need to deal with empty / missing handles, then perhaps the type of the `lambda_method_handle_map` should be `std::map<irep_idt, optionalt<lambda_method_handlest> >` instead of `std::map<irep_idt, lambda_method_handlest>`

Reply:
Or perhaps it isn't worth adding them to the map at all?

Reply:
best would probably to have a mapping from pairs of class name and entry number to the handle, then a non-existing (or non-parsable) lambda method handle would simply not exist in the map

Reply:
Yes, C++ doesn't make a large differentiation between the struct and class keywords. I am asking for a class to be used with each of the fields accessed through getter and setter methods instead of publicly accessible fields. The production of the debug text on lines 1480-1485, could then be defined as a method instead of inline at the point of building the class.